src/modules/timeTracker/timeTracker.redux.js

```diff
@@ -3,39 +3,42 @@ import Immutable from 'seamless-immutable';
 
 export const { Types: TimeTrackerTypes, Creators: TimeTrackerActions } = createActions(
   {
-    start: null,
+    start: ['benchmark'],
     stop: null,
+    reset: null,
     addSample: null,
-    saveResult: ['result'],
+    saveResult: ['benchmarkId', 'result'],
   },
   { prefix: 'TIME_TRACKER/' }
 );
 
 const INITIAL_STATE = Immutable({
   isActive: false,
-  startedAt: '0',
+  benchmark: null,
   samples: [],
-  duration: '0',
+  startedAt: 0,
 });
 
-const start = state =>
+const start = (state, benchmark) =>
   state.merge({
     isActive: true,
-    startedAt: '0',
+    benchmark: benchmark.benchmark,
+    startedAt: Date.now(),
     samples: [],
   });
 
 const stop = state =>
   state.merge({
     isActive: false,
-    startedAt: '0',
-    samples: [],
   });
 
-const addSample = state => state.set('samples', [...state.samples, { savedAt: String(Date.now()) }]);
+const addSample = state => state.set('samples', [...state.samples, { savedAt: Date.now() }]);
```

Comment:
```suggestion
const addSample = state => state.set('samples', state.samples.concat([{ savedAt: Date.now() }]));
```
Lepiej chyba tak, bo wraz ze wzrostem liczby sampli ten spread będzie się robić coraz wolniejszy.